### PR TITLE
Developing basic pipeline functionality 

### PR DESCRIPTION
**Pull request by kylacochrane**
This PR aims to develop the basic module requirements for the SpeciesAbundance Nextflow pipeline:

- fastp
- Kraken
- Bracken
- adjust_bracken_for_unclassified_reads
- Krona

Overall, it should adapt the previous [irida-plugin-species-abundance](https://github.com/Public-Health-Bioinformatics/irida-plugin-species-abundance) pipeline for Nextflow and IridaNext.

### CODE REVIEW

File: modules/local/fastptrim/main.nf [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    
+    if(meta.single_end || reads instanceof nextflow.processor.TaskPath) {
+        args = args + "-i ${reads[0]} -o ${meta.id}_trimmed.fastq.gz"
+    }else{
+        args = args + "-i ${reads[0]} -I ${reads[1]} -o ${meta.id}R1_trimmed.fastq.gz -O ${meta.id}R2_trimmed.fastq.gz"
```
Comment (apetkau) [inline modules/local/fastptrim/main.nf:31]: Could you add an underscore to the output file name. As in: `${meta.id}_R1_trimmed.fastq.gz`?

**Review by apetkau — CHANGES_REQUESTED** [review]
This looks amazing. Thanks so much @kylacochrane 😄 

In addition to the one comment below, could you also run the commands suggested in the github bot comment:

* Install [black](https://black.readthedocs.io/en/stable/): pip install black
* Fix formatting errors in your pipeline: black .

Some of the other github actions/tests we can work on fixing up later on.

File: modules/local/fastptrim/main.nf (hunk 1 above)
Comment (kylacochrane) [reply modules/local/fastptrim/main.nf]: No problem - updated in [7f8eb23](https://github.com/phac-nml/speciesabundance/pull/3/commits/7f8eb237ed497bc687246657943c1c05b1da0a00)

**Comment by kylacochrane** [comment]
- Fixed conflicts with Python linting (black) here: [7f8eb23](https://github.com/phac-nml/speciesabundance/pull/3/commits/7f8eb237ed497bc687246657943c1c05b1da0a00) 

- Then fixed up EditorConfig whitespace errors here: [dddba06](https://github.com/phac-nml/speciesabundance/pull/3/commits/dddba066da206fc852dea148f595acf52aefd96b)

File: modules/local/fastptrim/main.nf (hunk 1 above)
Comment (apetkau) [reply modules/local/fastptrim/main.nf]: Thanks so much 😄

**Review by apetkau — APPROVED** [review]
This looks great. Thanks so much 😄 

Approved. You can merge this in.

**Pull request merged**